### PR TITLE
[fix]: [CDS-85006]: add support for region field in aws connector

### PR DESCRIPTION
**Pull request by PragyeshMishra01**
## Describe your changes
1. Added support for aws region field in aws connector.
2. Added test cases for region field
## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`
- gitleaks: `trigger gitleaks`

### CODE REVIEW

**Comment by PragyeshMishra01** [comment]
trigger subcategoryfieldcheck

**Comment by PragyeshMishra01** [comment]
trigger subcategoryfieldcheck

File: .changelog/782.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:enhancement
```
Comment (RishabhGupta34) [inline .changelog/782.txt:1]: the file name has to be pr no.

File: docs/resources/platform_connector_aws.md [hunk 2]
```diff
@@ -110,6 +114,7 @@ resource "harness_platform_connector_aws" "aws" {
 - `org_id` (String) Unique identifier of the organization.
 - `project_id` (String) Unique identifier of the project.
 - `tags` (Set of String) Tags to associate with the resource.
+-  `region` AWS Region to perform Connection test of Connector.
```
Comment (RishabhGupta34) [inline docs/resources/platform_connector_aws.md:117]: is this auto-generated doc? As I see an extra space before region

File: docs/resources/platform_connector_aws.md [hunk 3]
```diff
@@ -162,6 +167,7 @@ Optional:
 Required:
 
 - `delegate_selectors` (Set of String) The delegates to inherit the credentials from.
+-  `region` AWS Region to perform Connection test of Connector.
```
Comment (RishabhGupta34) [inline docs/resources/platform_connector_aws.md:170]: same comment as above

File: docs/resources/platform_connector_aws.md [hunk 4]
```diff
@@ -170,6 +176,7 @@ Required:
...
```
Comment (RishabhGupta34) [inline docs/resources/platform_connector_aws.md:179]: same

File: docs/resources/platform_connector_aws.md [hunk 5]
```diff
@@ -184,6 +191,7 @@ Optional:
 - `access_key` (String) AWS access key.
 - `access_key_ref` (String) Reference to the Harness secret containing the aws access key. To reference a secret at the organization scope, prefix 'org' to the expression: org.{identifier}. To reference a secret at the account scope, prefix 'account` to the expression: account.{identifier}.
 - `delegate_selectors` (Set of String) Connect only use delegates with these tags.
+-  `region` AWS Region to perform Connection test of Connector.
```
Comment (RishabhGupta34) [inline docs/resources/platform_connector_aws.md:194]: same

File: .changelog/782.txt (hunk 1 above)
Comment (PragyeshMishra01) [reply .changelog/782.txt]: Done

File: docs/resources/platform_connector_aws.md (hunk 3 above)
Comment (PragyeshMishra01) [reply docs/resources/platform_connector_aws.md]: Fixed it

File: docs/resources/platform_connector_aws.md (hunk 4 above)
Comment (PragyeshMishra01) [reply docs/resources/platform_connector_aws.md]: Fixed it

File: docs/resources/platform_connector_aws.md (hunk 5 above)
Comment (PragyeshMishra01) [reply docs/resources/platform_connector_aws.md]: Fixed it

**Review by RishabhGupta34 — APPROVED** [review]

**Review by piyush-bhuwalka — APPROVED** [review]

**Pull request merged**